bundles/paikkatietoikkuna/coordinatetransformation/components/CoordinateSystemSelection.js

```diff
@@ -0,0 +1,241 @@
+            });
+            wrapper.append(coordinateSystemSelection);
+            this.setElement(wrapper);
+            this.populateSelect();
+            this.handleInfoLink();
+        },
+        /**
+         * @method populateSelect
+         * @desc creates an instance of the { Oskari.coordinatetransformation.component.select },
+         * and fills it with data
+         */
+        populateSelect: function () {
+            var me = this;
+            var wrapper = this.getElement();
+            this.select.create();
+            this.selectInstance = this.select.getSelectInstances();
+            this.dropdowns = this.select.getDropdowns();
+            Object.keys( this.dropdowns ).forEach( function( key ) {
+                jQuery( wrapper.find( '.transformation-system' ).find( me.makeClassSelector(key) ).append( me.dropdowns[key] ));
+            });
+            this.handleSelectionChanged();
+        },
+        handleInfoLink: function () {
+            var me = this;
+            this.getElement().find('.infolink').on('click', function ( event ) {
+                event.stopPropagation();
+                var key = this.parentElement.parentElement.dataset.system;
+                me.systemInfo.show( jQuery( this ), key );
+            });
+        },
+        handleSelectionChanged: function () {
+            var systems =  this.getElement().find('.system');
+            var me = this;
+            systems.each(function (index) {
+                jQuery( this ).on('change', function ( e ) {
+                    me.handleSystemChange(e);
+                });
+            })
+        },
+        handleSystemChange: function (e) {
+            var current = this.selectInstance[e.currentTarget.dataset.system];
+            var currentValue = current.getValue();
+            this.handleSelectValueChange(currentValue);
+        },
+        /**
+         * @method handleDropdownOptions
+         * @param {string} valueClass - class selector to show options for 
+         * @param {string} keyToEmpty - optional param to empty only one specific key in the dropdown 
+         */
+        handleDropdownOptions: function (valueClass, keyToEmpty) {
+            var dropdowns = this.dropdowns;
+            if ( valueClass.indexOf("DATUM_DEFAULT") !== -1 ) {
+                //show all options
+                Object.keys( dropdowns ).forEach( function ( key ) {
+                    dropdowns[key].find( 'option' ).show();
+                });
+                return;
+            }
+            if ( typeof keyToEmpty === 'undefined') {
+                Object.keys( dropdowns ).forEach( function ( key ) {
+                    dropdowns[key].find( 'option' ).hide();
+                    dropdowns[key].find( valueClass ).show();
+                });
+            } else {
+                Object.keys( dropdowns ).forEach( function ( key ) {
+                    dropdowns[keyToEmpty].find( 'option' ).hide();
```

Comment:
Looks like this does the same thing on each loop round. Can it be moved outside the loop like just before it?